src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -37,6 +36,9 @@ internal static class CustomModifierUtils
             // have already been compared.
             MethodSymbol constructedSourceMethod = sourceMethod.ConstructIfGeneric(destinationMethod.TypeArguments);
 
+            returnTypeCustomModifiers = constructedSourceMethod.ReturnTypeCustomModifiers;
+            countOfCustomModifiersPrecedingByRef = destinationMethod.ReturnsByRef ? constructedSourceMethod.CountOfCustomModifiersPrecedingByRef : (ushort)0;
```

Comment:
Why do we only conditionally invoke `constructedSourceMethod.CountOfCustomModifiersPrecedingByRef`? I assume there is a case where it throws / doesn't return 0 here but can't figure it out.